### PR TITLE
Documentation for the HTTP ingest

### LINKED ISSUE #1323: Add docs for multipart/mixed return
**Is your feature request related to a problem? Please describe.**

#1280 added support for returning transcoded data using `multipart/mixed` encoding when a broadcaster is using HTTP push, but there are no docs describing how to use the feature.

**Describe the solution you'd like**

We should add docs in `doc` directory describing how to use the `multipart/mixed` feature for a broadcaster using HTTP push. The docs can likely be as simple as describing the request and response objects.

**Describe alternatives you've considered**

We could add more context describing *why* someone would use the `multipart/mixed` feature, but that seems unnecessary right now. A guide for using `multipart/mixed` within a more complex system feels like its own separate thing.

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Adds documentation for the HTTP ingest

**Specific updates (required)**

- 
- 
- 

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

Fixes #1323 

**Checklist:**

- [ ] README and other documentation updated
- [ ] Node runs in OSX and devenv
- [ ] All tests in `./test.sh` pass

### CODE REVIEW

File: doc/ingest.md [hunk 1]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
 prior to transcoding. The stream can be pushed via a PUT or POST HTTP request to the
 `/live/` endpoint. HTTP request timeout is 8 seconds.
 
-The following are expected in the request:
+Body of the request should be binary data of the MPEG TS segment.
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
The body of the request should be the binary data of the MPEG TS segment.
```

File: doc/ingest.md [hunk 2]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
 
-* MPEG TS segments with segment numbers
+Two HTTP headers should be provided:
+  * `Content-Resolution` - in ther format `widthxheight`, for example: `1920x1080`.
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
  * `Content-Resolution` - in the format `widthxheight`, for example: `1920x1080`.
```

File: doc/ingest.md [hunk 3]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+  * `Content-Duration` - duration of segment, in milliseconds. Should be integer.
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
  * `Content-Duration` - duration of the segment, in milliseconds. Should be an integer.
```

File: doc/ingest.md [hunk 4]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+
+URL should have this structure:
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
The upload URL should have this structure:
```

File: doc/ingest.md [hunk 5]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+http://broadcasters:8935/live/movie/12.ts
+
+Where `movie` is name of the stream, `12` is the sequence number of the segment.
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
Where `movie` is name of the stream and `12` is the sequence number of the segment.
```

File: doc/ingest.md [hunk 6]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+
+HLS manifest will be available at
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
The HLS manifest will be available at 
```

File: doc/ingest.md [hunk 7]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+http://broadcasters:8935/live/movie/12.ts
+
+Where `movie` is name of the stream, `12` is the sequence number of the segment.
+
+HLS manifest will be available at 
+http://broadcasters:8935/stream/movie.m3u8
+
+Possble statuses returned by HTTP request:
+- 500 Internal Server Error - in case there was error during segment's transcode
+- 503 Service Unavailable - if broadcaster wasn't able to find orchestrator to transcode the segment
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
- 503 Service Unavailable - if the broadcaster wasn't able to find an orchestrator to transcode the segment
```

File: doc/ingest.md [hunk 8]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+- 200 OK - if transcoded successfully. Returned only after transcode completed 
+
+Optionally, actual transcoded segments or URLs to them can be returned in the response.
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
Optionally, actual transcoded segments or URLs pointing to them can be returned in the response.
```

File: doc/ingest.md [hunk 9]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+If `multipart/mixed` specified in the `Accept` header, then `Content-Type` of the response will be set to `multipart/mixed`, and body will consist of 'parts', according to [RFC1341](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html). If external object storage is used by broadcaster, then each part will have `application/vnd+livepeer.uri` content type, and will contain URL for the transcoded segment. If no external object storge used, then content type of each part will be `video/MP2T`, and body of the part will contain actual transcoded segment.
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
If `multipart/mixed` specified in the `Accept` header, then `Content-Type` of the response will be set to `multipart/mixed`, and the body will consist of 'parts', according to [RFC1341](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html). If external object storage is used by broadcaster, then each part will have `application/vnd+livepeer.uri` content type, and will contain URL for the transcoded segment. If no external object storage is used, then `Content-Type` of each part will be `video/MP2T`, and the body of the part will contain the actual transcoded segment.
```

File: doc/ingest.md [hunk 10]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+
+Each part will also contain such headers:
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
Each part will also contain these headers:
```

File: doc/ingest.md [hunk 11]
```diff
@@ -92,28 +92,84 @@ into the Livepeer network. Upon ingest, HTTP stream is pushed to the segmenter
...
+- `Content-Length` - length of the part
```
Comment (iameli) [inline doc/ingest.md]:
```suggestion
- `Content-Length` - length of the part in milliseconds
```

**Review by iameli — CHANGES_REQUESTED** [review]
Solid, just did a round of copyediting.

File: doc/ingest.md (hunk 11 above)
Comment (darkdarkdragon) [reply doc/ingest.md]: It is in bytes actually :)
fixed

**Comment by darkdarkdragon** [comment]
@iameli Thanks! Meaning of those articles still escapes me :(
fixed

**Review by iameli — APPROVED** [review]
LGTM!

**Pull request merged**